GrpcClient/GrpcRepo.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System.Runtime.InteropServices.JavaScript;
+using DTOs.Shift;
+using Google.Protobuf.WellKnownTypes;
+using Grpc.Net.Client;
+
+
+namespace GrpcClient;
+
+public class GrpcRepo
+{
+    public async Task<DTOs.EmployeeDTO> CreateEmployee(DTOs.NewEmployeeDTO newEmployeeDto)
+    {
+        //create channel for connection to JAVA
+        using var channel = GrpcChannel.ForAddress("http://192.168.125.143:50051"); //TODO the port might change
+        //create client for employee entity, "EmployeeClient" method is auto-genereted               
+        var client = new Employee.EmployeeClient(channel);
+        //reply - "AddSingleEmployee(newEmployeeDto)" comes from .proto file + need to convert the og-DTO to .proto-DTO
+        var reply =  client.AddSingleEmployee(new NewEmployeeDTO
+        {
+            FirstName = newEmployeeDto.FirstName,
+            LastName = newEmployeeDto.LastName,
+            WorkingNumber = UInt32.CreateChecked(newEmployeeDto.WorkingNumber),
+            Email = newEmployeeDto.Email,
+            Password = newEmployeeDto.Password
+        }); //TODO await maybe
+        //create new employee now with id to be returned to web app basically, source it from reply
+        EmployeeDTO employeeDto = new EmployeeDTO //maybe unnecessary
+        {
+            FirstName = reply.FirstName,
+            LastName = reply.LastName,
+            Id = reply.Id,
+            WorkingNumber = reply.WorkingNumber,
+            AssignedShifts = reply.AssignedShifts,
+            Email = reply.Email,
+            Password = reply.Password
+        };
+
+        List<ShiftDTO> grpcShiftDtos = employeeDto.AssignedShifts.Dtos.ToList();
+        List<DTOs.Shift.ShiftDTO> shiftDtos = new List<DTOs.Shift.ShiftDTO>();
+        grpcShiftDtos.ForEach((grpcShiftDto) =>
+        {
+            shiftDtos.Add(new DTOs.Shift.ShiftDTO
+            {
+                StartDateTime = new DateTime(grpcShiftDto.StartDateTime),
+                EndDateTime = new DateTime(grpcShiftDto.EndDateTime),
+                Description = grpcShiftDto.Description,
+                Id = grpcShiftDto.Id,
+                Location = grpcShiftDto.Location,
+                ShiftStatus = grpcShiftDto.ShiftStatus,
+                TypeOfShift = grpcShiftDto.ShiftStatus
+                
+            });
+        });
+        
+        //convert to og DTO
+        DTOs.EmployeeDTO ogEmployeeDto = new DTOs.EmployeeDTO
```

Comment:
what is og dto?

Reply:
og dto is Entity.Dto(s) and they are the original csharp project dtos